demo/alloy-showcase-webapp/src/test/java/com/liferay/faces/test/alloy/showcase/datatable/DataTableTester.java

```diff
@@ -0,0 +1,694 @@
+			" of 162 (Page " + pageNumber + " of 17)')]";
+	}
+
+	protected String dataTablePaginationLinkXpath(String whichPaginationActionLink) {
+		return listItemXpath("ul[@class='pagination']", "li") + "/*[contains(@onclick,\"'" + whichPaginationActionLink +
+			"'\")]";
+	}
+
+	protected String getTableRowCustomersXpath(String firstName, String lastName) {
+		return "//tbody/tr[contains(.,'" + firstName + "')][contains(.,'" + lastName +
+			"')]//input[contains(@id,'customers:')]";
+	}
+
+	protected String getTableRowXpath(String firstName, String lastName) {
+		return "//tbody/tr[contains(.,'" + firstName + "')][contains(.,'" + lastName + "')]";
+	}
+
+	protected String labelSelectOptionValueXpath(String labelText, String optionValue) {
+		return "//label[contains(.,'" + labelText + "')]/following-sibling::select/option[@value='" + optionValue +
+			"']";
+	}
+
+	protected void runDataTableLazyTest(String componentName) throws Exception {
+
+		Browser browser = Browser.getInstance();
+		navigateToUseCase(browser, componentName, "lazy");
+
+		testPagesPaginationDefault(browser);
+	}
+
+	protected void runDataTablePaginationTest(String componentName) throws Exception {
+
+		Browser browser = Browser.getInstance();
+		navigateToUseCase(browser, componentName, "pagination");
+
+		testPagesPaginationDefault(browser);
+
+		// Summary Position
+		// Test default "bottom"
+		SeleniumAssert.assertElementVisible(browser,
+			labelSelectOptionValueXpath("Summary Position", "bottom'][@selected='selected"));
+
+		String listItemResultsTag = "li/*[contains(.,'Results 1-10 of 162 (Page 1 of 17)'";
+		SeleniumAssert.assertElementVisible(browser,
+			listItemXpath("ul[@class='pagination'])[2]", listItemResultsTag) + "]");
+
+		// test select "top"
+		clickOptionAndWaitForAjaxRerender(browser, labelSelectOptionValueXpath("Summary Position", "top"));
+		browser.waitForElementPresent(labelSelectOptionValueXpath("Summary Position", "top'][@selected='selected"));
+		SeleniumAssert.assertElementVisible(browser,
+			listItemXpath("ul[@class='pagination'])[1]", listItemResultsTag) + "]");
+
+		// Test select "left"
+		clickOptionAndWaitForAjaxRerender(browser, labelSelectOptionValueXpath("Summary Position", "left"));
+		browser.waitForElementPresent(labelSelectOptionValueXpath("Summary Position", "left'][@selected='selected"));
+		SeleniumAssert.assertElementVisible(browser,
+			listItemXpath("ul[@class='pagination']", "li") + "[1]/*[contains(.,'Results 1-10 of 162 (Page 1 of 17)')]");
+
+		// Test select "right"
+		clickOptionAndWaitForAjaxRerender(browser, labelSelectOptionValueXpath("Summary Position", "right"));
+		browser.waitForElementPresent(labelSelectOptionValueXpath("Summary Position", "right'][@selected='selected"));
+		SeleniumAssert.assertElementVisible(browser,
+			"(//th[contains(.,'Customers')]/..//ul[@class='pagination']/li)[last()]/*[contains(.,'Results 1-10 of 162 (Page 1 of 17)')]");
```

Comment:
This could be simplified to:

```
SeleniumAssert.assertElementTextVisible(browser,
	"//th[contains(.,'Customers')]//ul[@class='pagination']/li[last()]/span", "Results 1-10 of 162 (Page 1 of 17)");
```

Please see if any other method calls aor xpath could be similarly simplified or made more readable.